pkg/broker/broker.go

```diff
@@ -297,6 +315,19 @@ func addNameAndIDForSpec(specs []*apb.Spec, registryName string) {
 		hasher := md5.New()
 		hasher.Write([]byte(spec.FQName))
 		spec.ID = hex.EncodeToString(hasher.Sum(nil))
+
+		// update the id on the plans, doing it here avoids looping through the
+		// specs array again
+		addIDForPlan(spec.Plans)
+	}
+}
+
+// addIDForPlan - for each of the plans create a new ID
+func addIDForPlan(plans []apb.Plan) {
+
+	// need to use the index into the array to actually update the struct.
+	for i := range plans {
+		plans[i].ID = uuid.New()
```

Comment:
What will happen here on restarting of the broker? Won't we blow away the spec and therefore get two different entries for the service plans much like what was happening with service instances names awhile ago and the reason we went to a hash of the FQname?

Reply:
Yes, that's an issue. Good catch. It seems we need the lookup ID to be a similar hash.

Reply:
UGH!